source/MaterialXContrib/MaterialXMaya/ShadingNodeOverrides.cpp

```diff
@@ -306,10 +308,25 @@ void ShadingNodeOverride<BASE>::updateShader(MHWRender::MShaderInstance& shaderI
     const mx::StringMap& inputs = ogsFragment->getPathInputMap();
     for (const auto& input : inputs)
     {
-        mx::ElementPtr element = document->getDescendant(input.first);
+        const std::string& elementPath = input.first;
+        mx::ElementPtr element = document->getDescendant(elementPath);
         if (!element)
         {
-            continue;
+            std::string nodePath = mx::parentNamePath(elementPath);
+            mx::ElementPtr uniformParent = document->getDescendant(nodePath);
+            if (uniformParent)
+            {
+                mx::NodePtr uniformNode = uniformParent->asA<mx::Node>();
+                if (uniformNode)
+                {
+                    mx::StringVec pathVec = mx::splitNamePath(elementPath);
+                    element = uniformNode->addInputFromNodeDef(pathVec[pathVec.size() - 1]);
```

Comment:
Just a check if you need to check the size of the returned vector? Not sure if the result could ever be 0 (which would make `pathVec[pathVec.size() - 1]` crash).